sechub-scan/src/main/resources/templates/report/html/scanresult.html

```diff
@@ -81,8 +81,51 @@
 			</table>
 		</div>
 		<div class="content">
+			<div>
+				<h2 class='summaryHeadline'>Summary</h2>
+				<table class='summaryTable'>
```

Comment:
The logic should be, that for each scan type you add a new column in the summary table.

For that you need to change the data structure of in the `HTMLScanResultReportModelBuilder`. The data structure should probably look like similar to this:

~~~

scanTypes:
| codeScan | { redList, yellowList, greenList } |
| secretScan | { redList, yellowList, greenList } |
| webScan | { redList, yellowList, greenList } |
| … | … |

~~~

Java Structure: `scanTypes = Map<String, List<<List<Findings>>>`

Add this to the Thymeleaf model: `model.put("scanTypes", scanTypes)`

then you can do something like this in pseudo code in the `scanresult.html`:

~~~
for scanType in ScanTypes:
    <th colspan="2" th:text="{scanType.name}">[scan type]</th>
~~~

and for the other entries:

~~~
for scanType in ScanTypes:
    <tr>
	 <td><strong>Total</strong></td>
        <td><strong><span th:text="*{#lists.size(scanType.redList) + #lists.size(scanType.yellowList) + #lists.size(scanType.greenList)}">[Total number of Elements]</span></strong></td>
	<td><strong></strong></td>
     </tr>
    …
~~~

This is roughly how it works. I probably got the `<tr>` and looping wrong.

For the second part of the overview with the CWE List you have to build out a similar structure, except this time you need to sort for CWE number:

~~~

scanTypesAndCwe:
| codeScan | { redList [ (78: 4), (84: 5) ], yellowList  [ (203: 324 ), (708: 523) ], greenList {} } |
| secretScan |  { redList [ (359: 1), (403: 500) ], yellowList  [ (13: 1), (903: 3) ], greenList {} } |
| webScan |  { redList, yellowList, greenList } |
| … | … |

~~~

The pair 78: 4 consists of two parts: cwe number, count.

This should be a structure: `Map<String, List<List<Map<Integer, Integer>>>>`.
You can use classes/objects to make your life easier.

The point is, that you have to prepare your data structures in such a way, that you are able to go through it in Thymeleaf.

Reply:
The alternative to using a lot of Lists and Maps is to create some helper classes.